standalone/src/DelphesHepMCReader.h

```diff
@@ -95,39 +95,42 @@ class DelphesHepMCInputReader : public DelphesInputReader {
                  TObjArray* partonOutputArray) override {
     m_treeWriter->Clear();
     m_readStopWatch.Start();
-    auto factory = modularDelphes->GetFactory();
+    auto factory  = modularDelphes->GetFactory();
+    bool finished = false;
     do {
-      m_finished = m_reader->ReadBlock(factory, allParticleOutputArray, stableParticleOutputArray, partonOutputArray);
-    } while (m_finished && !m_reader->EventReady());
+      finished = m_reader->ReadBlock(factory, allParticleOutputArray, stableParticleOutputArray, partonOutputArray);
+    } while (finished && !m_reader->EventReady());
```

Comment:
What about changing `finished` to another name since it just indicates whether the read was good or not, maybe something like:

```
bool goodRead;
while (goodRead = m_reader->ReadBlock(factory, allParticleOutputArray, stableParticleOutputArray, partonOutputArray) && !m_reader->EventReady()) {}
```